app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -52,11 +52,21 @@ def available_repositories
         end
       end
 
+      sorted_repos = repos.sort_by do |repo|
+        major, minor = repo[:substitutions][:releasever].nil? ? 1000.1000 : repo[:substitutions][:releasever].split('.').map(&:to_i)
```

Comment:
Sorry to add this late in the review, but is `Gem::Version` of any use here? We use this in other places when comparing versions

Reply:
No problem! I don't see any current examples in our code, but looking at what the class does I don't think it'd be a great fit since we are looking at things like architectures and versions that are not just things like '7.3' but also things like '7Server'. Not to say that it couldn't be made to work, but feels like we wouldn't get much value. Open to be shown otherwise though :)

Reply:
Sounds fair to me. I guess we don't use it in Katello, but I know its used in foreman-maintain since it deals with package versions, maybe that is where I was thinking of. Carry on :)